osmomath/decimal_test.go

```diff
@@ -148,6 +149,56 @@ func (s *decimalTestSuite) TestDecFloat64() {
 	}
 }
 
+func (s *decimalTestSuite) TestSdkDec() {
+	tests := []struct {
+		d    BigDec
+		want sdk.Dec
+		expPanic bool
+	}{
+		{NewBigDec(0), sdk.MustNewDecFromStr("0.000000000000000000"), false},
+		{NewBigDec(1), sdk.MustNewDecFromStr("1.000000000000000000"), false},
+		{NewBigDec(10), sdk.MustNewDecFromStr("10.000000000000000000"), false},
+		{NewBigDec(12340), sdk.MustNewDecFromStr("12340.000000000000000000"), false},
+		{NewDecWithPrec(12340, 4), sdk.MustNewDecFromStr("1.234000000000000000"), false},
+		{NewDecWithPrec(12340, 5), sdk.MustNewDecFromStr("0.123400000000000000"), false},
+		{NewDecWithPrec(12340, 8), sdk.MustNewDecFromStr("0.000123400000000000"), false},
+		{NewDecWithPrec(1009009009009009009, 17), sdk.MustNewDecFromStr("10.090090090090090090"), false},
```

Comment:
Doesn't this test case seem wrong?

Reply:
Definitely an odd case yes but it does seem correct to me (note that `NewDecWithPrec` shifts by 17 places, not 18 or 36). Since both the expected and actual values in each case pad zeroes for unused precision at the end these cases seem to be working as intended

Reply:
Err, isn't LHS: `1.009009009009009009`, and RHS `10.090090090090090090`, which is different values?

Reply:
Looks good to me. I think LHS is `10.09009009009009009` because of the precision of 17. RHS seems to only add an extra zero on the right since precision is 18

Reply:
@ValarDragon https://www.wolframalpha.com/input?i2d=true&i=Divide%5B1009009009009009009%2CPower%5B10%2C17%5D%5D
<img width="199" alt="Screen Shot 2022-09-19 at 8 00 03 AM" src="https://user-images.githubusercontent.com/62043214/191048786-305e5285-1a3f-4344-bdec-7aabfba6b3c9.png">